### DISCUSSION TITLE
Why CRC_USE_TABLE is not on by default?

### DISCUSSION THREAD

**Discussion by teksturi** [bacnet-stack/bacnet-stack#736] [General]
I do not understand why CRC_USE_TABLE is not on by default. I was first thinking that we try to save space when compiling to microcontrollers. Then looking makefiles we can see that arduino_uno, at91sam7s, atmega168, pic, and some others are only which enables this flag. Isn't table always faster than calculating that. Make no sense to me so please advice me why this is.

If there is no reason I would like to make it so that it is on by default (cmake option) and can be optionally set off.

**Comment by skarg** [comment]
Sometimes (seems often) I put defines (and other extras) in the example project Makefile to show that a library option exists. Since the CRC code comes directly from the BACnet standard, I would expect that code would be the default (and it is).

**Reply by teksturi** [reply]
I understand that standard would give example calculation which does not use table. However I would argue that most user wants to use table over pure calcutation. That is why I'm thinking can we add 

`CRC_USE_TABLE=1`

to cmake compile_definitions. We probably anyway should add definitions to there so we have one easy place where to change those. That is also good place to document those a little bit. Now user needs to read library code to find all defines and also think what ever they need those or not.
